editor/src/clj/editor/resource_node.clj

```diff
@@ -104,6 +115,15 @@
     (g/user-data-merge! user-data-values-by-key-by-node-id)
     (g/invalidate-outputs! invalidated-endpoints)))
 
+(g/defnk produce-lines [_node-id resource save-value]
+  (if (nil? save-value)
+    (resource-io/with-error-translation resource _node-id :lines
+      (let [content (slurp resource)]
+        (code.util/split-lines content)))
+    (let [resource-type (resource/resource-type resource)
+          content (save-content save-value resource-type)]
+      (code.util/split-lines content))))
+
```

Comment:
For `ResourceNodes` that output a `save-value`, we write it to a string and split it into lines so it can be displayed using the `CodeEditorView`. This means the lines will refresh whenever the resource is edited.

For `ResourceNodes` that don't output a `save-value`, we read the lines from the resource on disk (or in a zip). This might produce an `ErrorValue` in case the `resource` refers to a non-existing or unreadable resource. If the resource is modified due to external changes to the project, our `resource` argument dependency will be invalidated, and the view will refresh as a result.